daemon: Add new '--envoy-log' command line option and change the default path.

jrajahalme opened the pull request:
So far the Envoy logs were written to the state directory
(/var/run/cilium/cilium-envoy.log), the file system of which was
easily filled with Envoy debug logs, especially if something went
wrong.

Add a new cilium agent command line option to specify where Envoy logs
shoulld be written to. By default this will be
"/var/log/cilium-envoy.log", which is good for production runs. For
testing this default may need to be overridden with an explicit
commmand line option.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
Fixes: #2810

raybejjani reviewed (approved):
Excellent :) Thanks for doing this!

jrajahalme committed:
daemon: Add new '--envoy-log' command line option.

So far the Envoy logs were written to the state directory
(/var/run/cilium/cilium-envoy.log), the file system of which was
easily filled with Envoy debug logs, especially if something went
wrong.

Add a new cilium agent command line option to specify where Envoy logs
shoulld be written to. By default this will be
"/var/log/cilium-envoy.log", which is good for production runs. For
testing this default may need to be overridden with an explicit
commmand line option.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
